sonar-update-center-mojo/src/main/java/org/sonar/updatecenter/mojo/editions/EditionsGenerator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SonarSource :: Update Center :: Maven Plugin
+ * Copyright (C) 2010-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.updatecenter.mojo.editions;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.updatecenter.common.Plugin;
+import org.sonar.updatecenter.common.Release;
+import org.sonar.updatecenter.common.UpdateCenter;
+import org.sonar.updatecenter.common.Version;
+
+public class EditionsGenerator {
+
+  private static final Version MIN_SUPPORTED_SQ_VERSION = Version.create("6.5");
+  private static final Logger LOGGER = LoggerFactory.getLogger(EditionsGenerator.class);
+
+  private final UpdateCenter updateCenter;
+  private final EditionTemplatesLoader templatesLoader;
+  private final File jarsDir;
+
+  public EditionsGenerator(UpdateCenter updateCenter, EditionTemplatesLoader templatesLoader, File jarsDir) throws IOException {
+    this.updateCenter = updateCenter;
+    this.templatesLoader = templatesLoader;
+    this.jarsDir = jarsDir;
+    if (!jarsDir.exists()) {
+      throw new IllegalArgumentException("Directory " + jarsDir + " does not exist");
+    }
+  }
+
+  public void generateZipsAndJson(File outputDir, String downloadBaseUrl) throws IOException {
+    FileUtils.forceMkdir(outputDir);
+    FileUtils.cleanDirectory(outputDir);
+
+    try (FileWriter jsonWriter = new FileWriter(new File(outputDir, "editions.json"))) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "java.io.FileWriter" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1943)